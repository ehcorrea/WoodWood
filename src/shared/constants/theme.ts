import { rfvalue, rhvalue, rwvalue } from '../utils/responsive/responsive';

const COLORS_SCHEMES = {
  light: {
    primary: {
      main: '#EA5B0C',
      80: '#EF7A39',
      50: '#F4A273',
      20: '#F9D2B6',
    },
    secondary: {
      main: '#004ABE',
      80: '#336ECC',
      50: '#7FA5E0',
      20: '#CCD9F5',
    },
  },
};

export const THEME = {
  colors: {
    black: {
      main: '#1A1D26',
      80: '#2A2F3D',
      50: '#4D5364',
      20: '#6E7489',
    },
    white: {
      main: '#FFFFFF',
      80: '#FFFFFF',
      50: '#FFFFFF',
      20: '#FFFFFF',
    },
    gray: {
      main: '#9A9FAE',
      80: '#A8ACB9',
      50: '#C4C7D0',
      20: '#EBEBEB',
    },
    error: {
      main: '#F30000',
      80: '#F34C4C',
      50: '#F38585',
      20: '#F3BEBE',
    },
    transparent: {
      main: 'transparent',
      80: 'transparent',
      50: 'transparent',
      20: 'transparent',
    },
    ...COLORS_SCHEMES.light,
  },
  fonts: {
    weight: {
      light: 'SofiaPro-Light',
      regular: 'SofiaPro-Regular',
      medium: 'SofiaPro-Medium',
      semibold: 'SofiaPro-SemiBold',
      bold: 'SofiaPro-Bold',
    },
    size: {
      small: rfvalue(12).toString(),
      medium: rfvalue(15).toString(),
      large: rfvalue(18).toString(),
      xlarge: rfvalue(24).toString(),
      huge: rfvalue(30).toString(),
      xhuge: rfvalue(36).toString(),
    },
    height: {
      small: rfvalue(12).toString(),
      medium: rfvalue(18).toString(),
      large: rfvalue(20).toString(),
      xlarge: rfvalue(30).toString(),
      huge: rfvalue(50).toString(),
      xhuge: rfvalue(55).toString(),
    },
  },
  rfvalue,
  rwvalue,
  rhvalue,
} as const;

export type ThemeType = typeof THEME;
