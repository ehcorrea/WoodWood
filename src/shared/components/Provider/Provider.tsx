import { ThemeProvider } from '@emotion/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

import { THEME } from '../../constants/theme';

const queryClient = new QueryClient();

type ProviderProps = {
  children?: React.ReactElement;
};

export function Provider({ children }: ProviderProps) {
  return (
    <QueryClientProvider client={queryClient}>
      <ThemeProvider theme={THEME}>{children}</ThemeProvider>
    </QueryClientProvider>
  );
}
