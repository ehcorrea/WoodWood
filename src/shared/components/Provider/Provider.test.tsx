import { render } from '@testing-library/react-native';

import { THEME } from '../../constants/theme';
import { Text } from '../Text/Text';

import { Provider } from './Provider';

const MOCKED_TEXT = 'Mocked Text';

describe('<Provider/>', () => {
  describe('quando renderizado', () => {
    test('com children', () => {
      const { getByText, ...container } = render(
        <Provider>
          <Text>{MOCKED_TEXT}</Text>
        </Provider>
      );
      const text = getByText(MOCKED_TEXT);
      expect(text).toHaveStyle({ fontSize: Number(THEME.fonts.size.medium) });
      expect(container).toMatchSnapshot();
    });
    test('sem children', () => {
      const container = render(<Provider />);
      expect(container).toMatchSnapshot();
    });
  });
});
