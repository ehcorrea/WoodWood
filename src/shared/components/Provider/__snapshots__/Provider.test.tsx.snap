// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Provider/> quando renderizado com children 1`] = `
<Text
  style={
    [
      {
        "color": "#1A1D26",
        "fontFamily": "SofiaPro-Regular",
        "fontSize": 21.5,
        "lineHeight": 25.5,
      },
      undefined,
    ]
  }
>
  Mocked Text
</Text>
`;

exports[`<Provider/> quando renderizado sem children 1`] = `null`;
